Modules/pyexpat.c

```diff
@@ -430,45 +442,57 @@ my_StartElementHandler(void *userData,
     }
 }
 
-#define RC_HANDLER(RC, NAME, PARAMS, INIT, PARAM_FORMAT, CONVERSION, \
-                RETURN, GETUSERDATA) \
-static RC \
-my_##NAME##Handler PARAMS {\
-    xmlparseobject *self = GETUSERDATA ; \
-    PyObject *args = NULL; \
-    PyObject *rv = NULL; \
-    INIT \
-\
-    if (have_handler(self, NAME)) { \
-        if (PyErr_Occurred()) \
-            return RETURN; \
-        if (flush_character_buffer(self) < 0) \
-            return RETURN; \
-        args = Py_BuildValue PARAM_FORMAT ;\
-        if (!args) { flag_error(self); return RETURN;} \
-        self->in_callback = 1; \
-        rv = call_with_frame(#NAME,__LINE__, \
-                             self->handlers[NAME], args, self); \
-        self->in_callback = 0; \
-        Py_DECREF(args); \
-        if (rv == NULL) { \
-            flag_error(self); \
-            return RETURN; \
-        } \
-        CONVERSION \
-        Py_DECREF(rv); \
-    } \
-    return RETURN; \
+#define RC_HANDLER(RETURN_TYPE, NAME, PARAMS,       \
+                   INIT, PARSE_FORMAT, CONVERSION,  \
+                   RETURN_VARIABLE, GETUSERDATA)    \
```

Comment:
RETURN_VARIABLE is not always a variable.

Reply:
Oh, never mind, this PR changes that!

Reply:
Yes, because reading the macro was really painful and I took the liberty of changing them in this PR. Do you actually prefer a separate PR for this?

Reply:
> Do you actually prefer a separate PR for this?

Very slightly, yes. Not worth the overhead of producing such a PR.